tap_hubspot/hubspot.py

```diff
@@ -406,41 +471,31 @@ def get_companies(
                 self.get_value(company, ["properties", filter_key])
             )
 
-    def get_contacts(
-        self, start_date: datetime, end_date: datetime
-    ) -> Iterable[Tuple[Dict, datetime]]:
-        self.event_state["contacts_start_date"] = start_date
-        self.event_state["contacts_end_date"] = end_date
-        filter_key = "lastmodifieddate"
-        obj_type = "contacts"
-        properties = self.get_object_properties(obj_type)
+    def get_contacts(self) -> Iterable[Tuple[Dict, datetime]]:
+        replication_path = ["updatedAt"]
 
-        gen = self.search(
-            obj_type,
-            filter_key,
-            start_date,
-            end_date,
-            properties,
+        gen = self.get_records(
+            "/crm/v3/objects/contacts",
+            replication_path,
+            params={"limit": 100, "properties": MANDATORY_PROPERTIES["contacts"]},
+            data_field="results",
+            offset_key="after",
         )
 
         for chunk in chunker(gen, 100):
-            ids: List[str] = [contact["id"] for contact in chunk]
-
-            companies_associations = self.get_associations(obj_type, "companies", ids)
-
-            for i, contact_id in enumerate(ids):
-                contact = chunk[i]
-
-                companies = companies_associations.get(contact_id, [])
+            ids: List[str] = [contact["id"] for contact, _ in chunk]
```

Comment:
It seems like you try to unpack `chunk` to 2 vars. But `chunk` comes from `chunker(.)` that returns a list[T] (`def chunker(iter: Iterable[T], size: int) -> Iterable[List[T]]:`).

I was expecting `[contact["id"] for contact in chunk]` instead.

Reply:
ah, gotcha, generic T infered from the generator 👍